File: pset1/cash/cash.c
#include <stdio.h>
#include <math.h>
#include <cs50.h>

int main(void)
{
    float input = 0;
    // get valid input
    do
    {
        input = get_float("Change Owed: ");
    }
    while (input <= 0);
    // round the input cents to dollars
    int cents = round(input * 100);
    // final number of coins
    int result = 0;
    // check from the most largest cents to the smallest
    while (cents > 0)
    {
        if (cents >= 25)
        {
            cents -= 25;
        }
        else if (cents >= 10)
        {
            cents -= 10;
        }
        else if (cents >= 5)
        {
            cents -= 5;
        }
        else
        {
            cents -= 1;
        }
        // each loop will increase one coin
        result++;
    }
    printf("%i\n", result);
}

File: pset6/cash/cash.py
from cs50 import get_float
# keep looking for positive input
dollar = 0
while (dollar <= 0):
    dollar = get_float("Change owed: ")
    
total = 0
cent = round(dollar * 100, 0)

while (cent > 0):
    # 0.25
    if (cent >= 25):
        cent -= 25
    # 0.1
    elif (cent >= 10):
        cent -= 10
    # 0.05
    elif (cent >= 5):
        cent -= 5
    # 0.01
    else:
        cent -= 1
    total += 1
    
# output
print(f"{total}")